onmt/transforms/tokenize.py

```diff
@@ -265,7 +262,14 @@ class ONMTTokenizerTransform(TokenizerTransform):
     def __init__(self, opts):
         """Initialize necessary options for OpenNMT Tokenizer."""
         super().__init__(opts)
-        self._parse_opts()
+
+    def _set_seed(self, seed):
+        """set seed to ensure reproducibility."""
+        try:
+            import pyonmttok
+            pyonmttok.set_random_seed(seed)
+        except AttributeError as err:
```

Comment:
You could increase the required version of the package in `setup.py` instead of checking for feature availability.

Reply:
Thanks for reminding me, just realize we've already made the restriction on the version of those packages. I'll change this.